tests/main/interfaces-contacts-service/task.yaml

```diff
@@ -51,9 +47,14 @@ execute: |
     fi
     mkdir -p "$XDG_CONFIG_HOME" "$XDG_DATA_HOME" "$XDG_CACHE_HOME"
 
-    echo "Setting up D-Bus session bus"
-    eval "$(dbus-launch --sh-syntax)"
-    echo "$DBUS_SESSION_BUS_PID" > dbus-launch.pid
+    echo "Setting up D-Bus session bus in a systemd unit"
+    systemd-run --unit=dbus-session --property=Type=forking -E "$(env | tr '\n' ' ')"  -r /bin/sh -c 'dbus-launch --sh-syntax > /tmp/dbus-sh'
```

Comment:
What is the environment part for? Can you explain what do we need from the local environment to make the service operate the way we want?

In addition, can the service run `exec dbus-launch` instead, to make sure there's just enough processes as necessary, not more.

Reply:
Its needed for the XDG_* vars.

Reply:
I think it warrants a comment. We could even constrain it to just `XDG_*=*` for clarity

Reply:
I had to change this as older versions of systemd do not support "-E" - now its explicit.